components/core/src/clp_s/search/ColumnDescriptor.hpp

```diff
@@ -56,6 +56,13 @@ class DescriptorToken {
      */
     std::string const& get_token() const { return m_token; }
 
+    /**
+     * Equal to operator to allow comparison between descriptor tokens.
+     * @return true if this token is equal to right hand side
+     * @return false if this token is not equal to the right hand side
```

Comment:
```suggestion
     * @return Whether this token is equal to the given token
```

* Alternatively, I think we could drop this docstring since it doesn't add much.
* Minor, but you could add a comment in the method saying that `m_wildcard` and `m_regex` are properties derived from `m_token`, so that's why we only need to compare `m_token`.